src/core/opengl.cpp

```diff
@@ -705,7 +705,7 @@ void program_t::free_resources()
 {
     for (int i = 0; i < wf::TEXTURE_TYPE_ALL; i++)
     {
-        if (this->priv->id[i])
+        if (this->priv && this->priv->id[i])
```

Comment:
This sounds like a workaround for a race condition which happens somewhere else. free_resources() should never be called if this->priv is NULL - that would mean that we call the function after the program object has been freed.